website/maintenance.py

```diff
@@ -40,27 +29,31 @@ def set_maintenance(start=None, end=None):
         start = end - timedelta(1)
 
     unset_maintenance()
-    # NOTE: We store isoformatted dates in order to preserve timezone information (pymongo retrieves naive datetimes)
-    database.maintenance.insert({'maintenance': True, 'start': start.isoformat(), 'end': end.isoformat()})
+
+    state = MaintenanceState.objects.create(
+        start=start,
+        end=end
+    )
+
+    return {'start': state.start, 'end': state.end}
 
 
 def get_maintenance():
-    """Get the current start and end times for the maintenance state.
-    Return None for start and end if there is no maintenance state
+    """Get the current start and end times for the current maintenance state.
+    Return None for start and end if there is no current maintenance state.
+    Return None if there is a maintenance state set for the future.
     """
-    if not database:
-        return None
-    maintenance_state = database.maintenance.find_one({'maintenance': True})
+    now = timezone.now()
+    maintenance_state = MaintenanceState.objects.filter(start__lte=now, end__gte=now).first()
```

Comment:
I wanted to be able to set maintenance periods in the future -- this way there can be a state set for the future, and then `get_maintenance` will still return the right thing if there isn't maintenance at that exact moment.  I can change it tho so that you can't set periods for the future if that'll be better!